### DISCUSSION TITLE
📚 Doc: Example of `after_hook`

### DISCUSSION THREAD

**Discussion by tgsmith61591** [cucumber-rs/cucumber#225] [Q&A] [answered]
I'm having a really difficult time understanding how to add an `after_hook` to my `World` struct. Is there an example in the documentation you can point me to? [This is the closest](https://docs.rs/cucumber/latest/cucumber/struct.Cucumber.html#method.after) I can find, and the [linked source](https://docs.rs/cucumber/latest/src/cucumber/cucumber.rs.html#1102) is not too helpful either.

Assuming I have a world as follows:

```rust
#[async_trait(?Send)]
impl World for MyCoolWorld {
    type Error = Infallible;

    async fn new() -> Result<Self, Infallible> {
        Ok(Self {
            thing_one: None,
            thing_two: None,
        })
    }
}
```
And I'm running like so:

```rust
    MyCoolWorld::cucumber()
        .run_and_exit("./products")
        .await;
```
How/where do I attach an `after` hook? Is it in the `World` impl block? Is it a builder when I set up the runner? Any example you can provide would be extremely appreciated!

**Comment by ilslv** [comment, marked as answer]
We have a [separate book chapter](https://cucumber-rs.github.io/cucumber/main/writing/hooks.html#after-hook) covering `Scenario` hooks. In your case it would look something like

```rust
MyCoolWorld::cucumber()
    .after(|feature, rule, scenario, world| {
        // Hook logic here.
    })
    .run_and_exit("./products")
    .await;
```

**Reply by tgsmith61591** [reply]
Thanks! Easy and straight forward. Did not find that chapter before
